src/utils/deprecated/deprecated.tsx

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+
+export const getDeprecatedMessage = (message: string): string =>
+  `[DEPRECATED] ${message}`;
+
+export const deprecated = (message: string) => {
+  return function <T extends React.ComponentType<any>>(Component: T): T {
+    const deprecatedWrapper = (props: React.ComponentProps<T>) => {
+      console.warn(getDeprecatedMessage(message));
```

Comment:
Should this be wrapped in a `useEffect(..., []);` to prevent the message from appearing multiple times?

Reply:
Good idea, implemented it!